lib/MySQL_Thread.cpp

```diff
@@ -4091,6 +4091,128 @@ void MySQL_Thread::unregister_session(int idx) {
 }
 
 
+// this function was inline in MySQL_Thread::run()
+void MySQL_Thread::run___get_multiple_idle_connections(int& num_idles) {
+	int rc;
+	int i;
+	num_idles=MyHGM->get_multiple_idle_connections(-1, curtime-mysql_thread___ping_interval_server_msec*1000, my_idle_conns, SESSIONS_FOR_CONNECTIONS_HANDLER);
+	for (i=0; i<num_idles; i++) {
+		MySQL_Data_Stream *myds;
+		MySQL_Connection *mc=my_idle_conns[i];
+		MySQL_Session *sess=new MySQL_Session();
+		sess->mybe=sess->find_or_create_backend(mc->parent->myhgc->hid);
+
+		myds=sess->mybe->server_myds;
+		myds->attach_connection(mc);
+		myds->assign_fd_from_mysql_conn();
+		myds->myds_type=MYDS_BACKEND;
+
+		sess->to_process=1;
+		myds->wait_until=curtime+mysql_thread___ping_timeout_server*1000;	// max_timeout
+		mc->last_time_used=curtime;
+		myds->myprot.init(&myds, myds->myconn->userinfo, NULL);
+		sess->status=PINGING_SERVER;
+		myds->DSS=STATE_MARIADB_PING;
+		register_session_connection_handler(sess,true);
+		int rc=sess->handler();
+		if (rc==-1) {
+			unsigned int sess_idx=mysql_sessions->len-1;
+			unregister_session(sess_idx);
+			delete sess;
+		}
+	}
+	processing_idles=true;
+	last_processing_idles=curtime;
+}
+
+// this function was inline in MySQL_Thread::run()
+void MySQL_Thread::ProcessAllMyDS_BeforePoll() {
+	unsigned int n;
+	for (n = 0; n < mypolls.len; n++) {
+		MySQL_Data_Stream *myds=NULL;
+		myds=mypolls.myds[n];
+		mypolls.fds[n].revents=0;
+		if (myds) {
+#ifdef IDLE_THREADS
+			if (GloVars.global.idle_threads) {
+				// here we try to move it to the maintenance thread
+				if (myds->myds_type==MYDS_FRONTEND && myds->sess) {
+					if (myds->DSS==STATE_SLEEP && myds->sess->status==WAITING_CLIENT_DATA) {
+						if (move_session_to_idle_mysql_sessions(myds, n)) {
+							n--;  // compensate mypolls.remove_index_fast(n) and n++ of loop
+							continue;
+						}
+					}
+				}
+			}
+#endif // IDLE_THREADS
+			if (unlikely(myds->wait_until)) {
+				tune_timeout_for_myds_needs_pause(myds);
+			}
+			if (myds->sess) {
+				if (unlikely(myds->sess->pause_until > 0)) {
+					tune_timeout_for_session_needs_pause(myds);
+				}
+			}
+			myds->revents=0;
+			if (myds->myds_type!=MYDS_LISTENER) {
+				configure_pollout(myds, n);
+			}
+		}
+		proxy_debug(PROXY_DEBUG_NET,1,"Poll for DataStream=%p will be called with FD=%d and events=%d\n", mypolls.myds[n], mypolls.fds[n].fd, mypolls.fds[n].events);
+	}
+}
+
+
+// this function was inline in MySQL_Thread::run()
+void MySQL_Thread::ProcessAllMyDS_AfterPoll() {
+	int rc;
```

Comment:
Never used, replaced locally in line `4195`.

Reply:
Fxing